### PR TITLE
Apply PlonePAS patch for delOb

### PR DESCRIPTION
**Pull request by icemac**
According to plone/Products.PluggableAuthService#3 this might be valuable upstream to be able to bury the plone fork.

The most important part seems to be 7d6b40d.

### CODE REVIEW

**Review by mauritsvanrees — APPROVED** [review]
LGTM.
I don't agree with every pep8 changes, some introduce other pep8 violations, but it is an improvement, so I am okay with this.

File: Products/PluggableAuthService/PluggableAuthService.py [hunk 1]
```diff
@@ -437,15 +437,21 @@ def _setObject(self, id, object, roles=None, user=None, set_owner=0):
...
+        # existence, since it replaces objects (like 'plugins') with 
+        # SimpleItems and calls _delOb, which tries to use special methods
+        # of 'plugins'
+
+        # XXX imo this is a evil one
+
         plugins = self._getOb('plugins', None)
 
-        if plugins is not None:
+        if getattr(plugins, 'removePluginById', None) is not None:
```
Comment (tseaver) [inline Products/PluggableAuthService/PluggableAuthService.py]: Why are we modifying this code here?  ZVC shouldn't be replacing a PAS instance's `plugins` with something which does not implement `Products.PluginRegistry.interfaces.IPluginRegistry`.

**Review by tseaver — CHANGES_REQUESTED** [review]
Mixing in a huge pile of PEP-8 / janitorial changes with a substantive change isn't great for the review process:  please keep them separate in the future.

**Comment by tseaver** [comment]
@icemac Issue #3 seems unrelated to this PR:  can you please clarify?

**Comment by icemac** [comment]
@tseaver This PR actually refers to plone/Products.PluggableAuthService#3, sorry it was my fault.

File: Products/PluggableAuthService/PluggableAuthService.py (hunk 1 above)
Comment (icemac) [reply Products/PluggableAuthService/PluggableAuthService.py]: @esteele Can you please explain the reason for this change in more detail as you committed it in the first place?

Comment (esteele) [reply Products/PluggableAuthService/PluggableAuthService.py]: I just pulled it from the PlonePAS patch. That change is at least 11 years old. 
https://github.com/plone/Products.PlonePAS/blame/1d8f9263b2ab6df93d67d5fbf66d49527c57f57a/src/Products/PlonePAS/pas.py#L374

Comment (icemac) [reply Products/PluggableAuthService/PluggableAuthService.py]: @esteele Was it ever used in a release or in production or can we drop this PR as irrelevant?

Comment (mauritsvanrees) [reply Products/PluggableAuthService/PluggableAuthService.py]: The patch in PlonePAS has been in use for the past eleven years, and is happily used by any non-stone-age Plone site. Merging the patch upstream in PluggableAuthService would mean we can remove a patch from PlonePAS.
This particular check seems harmless.

Comment (dataflake) [reply Products/PluggableAuthService/PluggableAuthService.py]: I concur with Tres. Why can't those things that ZVC uses as ``plugins`` grow a ``removePluginById`` method? Special cases like that don't really belong upstream.

**Review by dataflake — CHANGES_REQUESTED** [review]
The PR cannot be used as-is at this point due to the PEP-8 changes. I went through the master branch yesterday and today and made it all PEP-8 compatible. See f75797e375118e32096bbd5211e4e7ea13c4219f

**Comment by icemac** [comment]
As suggested by @tseaver and @dataflake this problem should be fixed downstream.
Closing this PR.

**Pull request closed without merging**